dsymbol/src/dsymbol/conversion/package.d

```diff
@@ -115,7 +115,7 @@ class AutocompleteParser : Parser
 	{
 		if (!currentIs(tok!"{"))
 			return null;
-		if (current.index > cursorPosition)
+		if (cursorPosition != -1 && current.index > cursorPosition)
```

Comment:
```suggestion
		if (cursorPosition == -1) return super.parseBlockStatement();
		if (current.index > cursorPosition)
```

I'm thinking that's all that's needed, no? Otherwise the code always sets a bookmark, skips braces and then immediately rolls back again.